### PR TITLE
For #1346: Make correct zold payment.

### PR DESCRIPTION
**Pull request by paulodamaso**
For #1346: Make correct zold payment:
- Added bundle test for zold payment to user with vesting and user without vesting;
- Added puzzle to direct alterations in `send_zold.groovy`

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1646](https://github.com/zerocracy/farm/issues/1646) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1646](https://github.com/zerocracy/farm/issues/1646) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/2360523645); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/resources/com/zerocracy/bundles/send_correct_zold_value/_after.groovy [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+def exec(Project project, XML xml) {
+//  Farm farm = binding.variables.farm
+//  new Footprint(farm, project).withCloseable { Footprint footprint ->
+//    MatcherAssert.assertThat(
+//      'User with vesting received wrong ZLD value',
+//      footprint.collection().find(
+//        Filters.and(
+//          Filters.eq('type', 'Notify test'),
+//          Filters.eq('login', 'krzyk'),
+//          Filters.eq('message', 'We just sent you 100 ZLD through https://wts.zold.io'),
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/send_correct_zold_value/_after.groovy]: @paulodamaso I think this should be 120 (cash rate + vesting rate)

File: src/test/resources/com/zerocracy/bundles/send_correct_zold_value/claims.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+-->
+<claims xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://datum.zerocracy.com/0.61.2/xsd/pm/claims.xsd" updated="2017-07-21T20:31:44.512Z" version="1">
+  <claim id="1">
+    <created>2017-07-21T20:32:19.716Z</created>
+    <type>Make payment</type>
+    <author>0crat</author>
+    <token>test;C123;user42</token>
+    <params>
+      <param name="login">amihaiemil</param>
+      <param name="minutes">30</param>
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/send_correct_zold_value/claims.xml]: @paulodamaso this should be 60 minutes (or you need to change the _after value to 10 ZLD) and cash below $20

File: src/test/resources/com/zerocracy/bundles/send_correct_zold_value/claims.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    </params>
+  </claim>
+  <claim id="2">
+    <created>2017-07-21T20:32:19.716Z</created>
+    <type>Make payment</type>
+    <author>0crat</author>
+    <token>test;C123;user43</token>
+    <params>
+      <param name="login">krzyk</param>
+      <param name="minutes">30</param>
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/send_correct_zold_value/claims.xml]: @paulodamaso similarly here, 60 minutes and $20 (unless _after values change to 60 ZLD)

File: src/test/resources/com/zerocracy/bundles/send_correct_zold_value/_after.groovy [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+//    MatcherAssert.assertThat(
+//      'User without vesting received wrong ZLD value',
+//      footprint.collection().find(
+//        Filters.and(
+//          Filters.eq('type', 'Notify test'),
+//          Filters.eq('login', 'amihaiemil'),
+//          Filters.eq('message', 'We just sent you 20 ZLD through https://wts.zold.io'),
+//        )
+//      ),
+//      new IsIterableWithSize<>(1)
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/send_correct_zold_value/_after.groovy]: @paulodamaso should be `new IsIterableWithSize<>(new IsEqual<>(1)) `

File: src/test/resources/com/zerocracy/bundles/send_correct_zold_value/_after.groovy [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+//    MatcherAssert.assertThat(
+//      'User with vesting received wrong ZLD value',
+//      footprint.collection().find(
+//        Filters.and(
+//          Filters.eq('type', 'Notify test'),
+//          Filters.eq('login', 'krzyk'),
+//          Filters.eq('message', 'We just sent you 100 ZLD through https://wts.zold.io'),
+//        )
+//      ),
+//      new IsIterableWithSize<>(1)
```
Comment (krzyk) [inline src/test/resources/com/zerocracy/bundles/send_correct_zold_value/_after.groovy]: @paulodamaso should be `new IsIterableWithSize<>(new IsEqual<>(1)) `

**Comment by paulodamaso** [comment]
@krzyk Thanks for the comments, I did not noticed that zold calculation error. Take a look at the fixes.

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1646](https://github.com/zerocracy/farm/issues/1646) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**